source/custom/rules/pci_id_rule.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules
+
+import (
+	"fmt"
+	pciutils "sigs.k8s.io/node-feature-discovery/source/internal"
+)
+
+// Rule that matches on the following PCI device attributes: <class, vendor, device>
+// each device attribute will be a list elements(strings).
+// Match operation: OR will be performed per element and AND will be performed per attribute.
+// An empty attribute will not be included in the matching process.
+type PciIdRuleInput struct {
+	Class  []string `json:"class,omitempty"`
+	Vendor []string `json:"vendor,omitempty"`
+	Device []string `json:"device,omitempty"`
+}
+
+type PciIdRule struct {
+	PciIdRuleInput
```

Comment:
Is there a reason to have the intermediate `PciIdRuleInput` type? Could it be dropped and just define the members (Class, Vendor, Device) here?

Reply:
its just so we have clear boundary between whats being provided by the user via config and and possibly additional (internal) attributes added to the rule .

I wanted to add that for the LoadedKMod rule as well but as the rule is simple it would make the user config more complex.

Reply:
> its just so we have clear boundary between whats being provided by the user via config and and possibly additional (internal) attributes added to the rule .

OK. You could do a similar separation with public/non-public members(?) But, let's keep this if you prefer the really tight separation.